lib/datadog/statsd/serialization/stat_serializer.rb

```diff
@@ -38,7 +38,7 @@ def global_tags
         attr_reader :tag_serializer
 
         def formated_name(name)
-          if name.is_a?(String)
+          if name.is_a?(String) && name.include?(':')
```

Comment:
I think we need to be more careful about this.

The old code guaranteed that a copy of `name` is made before any modifying operations like `#gsub!` or `#tr!` are called, and that's super important because otherwise re-using a metric name is no longer safe.

The change you're proposing here will cause metric names that don't include a `:` to go into the `else` branch, which in turn will call `#to_s`, which will return `name`, and then we will end up calling `#gsub!` and `#tr!` on the `name` object, modifying it in place, breaking the guarantee this had before.

I think you want to change this like this:

```ruby
  formated = name.to_s unless name.is_a?(String)

  if formated.include?('::')
    formated = formated.gsub('::', '.')
    formated.tr!(':|@', '_')
    formated
  elsif formated.include?(':') || formated.include?('@') || formated.include?('|')
    formated.tr(':|@', '_')
  else
    formated
  end
```

Reply:
Not sure if using 3 separete `include?` checks is great - we might want to use a regexp instead.